python/graphstorm/config/argument.py

```diff
@@ -1874,9 +1874,128 @@ def lp_edge_weight_for_loss(self):
 
         return None
 
+    @property
+    def train_hard_edge_dstnode_negative(self):
+        """ The list of canonical etypes that have hard negative sets
+
+            The format of the arguement should be:
+            train_hard_edge_dstnode_negative:
+              - src_type,rel_type0,dst_type:negative_nid_field
+              - src_type,rel_type1,dst_type:negative_nid_field
+            Each edge type can have different fields storing the hard negatives.
+
+            or
+            train_hard_edge_dstnode_negative:
+              - negative_nid_field
+            All the edge types use the same filed storing the hard negatives.
+        """
+        # pylint: disable=no-member
+        if hasattr(self, "_train_hard_edge_dstnode_negative"):
+            assert self.task_type == BUILTIN_TASK_LINK_PREDICTION, \
+                "Hard negative only works with link prediction"
+            hard_negatives = self._train_hard_edge_dstnode_negative
+            if len(hard_negatives) == 1 and \
+                ":" not in hard_negatives[0]:
+                # global feat_name
+                return hard_negatives[0]
+
+            # per edge type feature
+            hard_negative_dict = {}
+            for hard_negative in hard_negatives:
+                negative_info = hard_negative.split(":")
```

Comment:
do we need to verify the correctness of the input format?

Reply:
Add some.